pkg/appgw/configbuilder.go

```diff
@@ -179,3 +179,31 @@ func (c *appGwConfigBuilder) addTags() {
 	// Identify the App Gateway as being exclusively managed by a Kubernetes Ingress.
 	c.appGw.Tags[managedByK8sIngress] = to.StringPtr(fmt.Sprintf("%s/%s/%s", version.Version, version.GitCommit, version.BuildDate))
 }
+
+func (c *appGwConfigBuilder) getExistingBackendPools() []n.ApplicationGatewayBackendAddressPool {
+	if c.appGw.BackendAddressPools == nil {
+		return []n.ApplicationGatewayBackendAddressPool{}
+	}
+	return *c.appGw.BackendAddressPools
+}
+
+func (c *appGwConfigBuilder) getExistingListeners() []n.ApplicationGatewayHTTPListener {
+	if c.appGw.HTTPListeners == nil {
+		return []n.ApplicationGatewayHTTPListener{}
+	}
+	return *c.appGw.HTTPListeners
+}
+
+func (c *appGwConfigBuilder) getExistingRoutingRules() []n.ApplicationGatewayRequestRoutingRule {
+	if c.appGw.BackendAddressPools == nil {
```

Comment:
s/BackendAddressPools/ApplicationGatewayRequestRoutingRule

Reply:
Thanks for catching this!